clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -848,6 +849,42 @@ private void timeoutCallsInFlight(TimeoutProcessor processor, Map<String, List<C
         }
 
         /**
+         * If an authentication exception is encountered with connection to any broker,
+         * fail all pending requests. Returns true if an authentication exception was encountered.
+         */
+        private boolean handleAuthenticationException(long now, Map<Node, List<Call>> callsToSend) {
+            AuthenticationException authenticationException = null;
+            try {
+                metadata.maybeThrowAuthenticationException();
+            } catch (AuthenticationException e) {
+                authenticationException = e;
+            }
+            if (authenticationException == null) {
+                for (Node node : callsToSend.keySet()) {
+                    authenticationException = client.authenticationException(node);
+                    if (authenticationException != null)
+                        break;
+                }
+            }
+            if (authenticationException != null) {
+                synchronized (this) {
+                    for (Call newCall : newCalls) {
+                        newCall.fail(now, authenticationException);
+                    }
+                    newCalls.clear();
+                }
+                for (List<Call> calls : callsToSend.values()) {
+                    for (Call call : calls) {
+                        call.handleFailure(authenticationException);
```

Comment:
Why are we calling `handleFailure` directly` instead of `fail`? If we used `fail` in both places, we could extract a method and call it twice in succession, once for `newCalls` and once for `callsToSend`.